upstream bug: http://bugs.php.net/bug.php?id=46311
The current patch follows a suggestion from Siarhei Siamashka
--- php-5.3.3.orig/Zend/zend_execute.h	2010-07-26 00:11:41.000000000 +0200
+++ php-5.3.3/Zend/zend_execute.h	2010-07-26 00:12:30.000000000 +0200
@@ -313,7 +313,7 @@
 	int delete_count = (int)(zend_uintptr_t) *p;
 
 	while (--delete_count>=0) {
-		zval *q = *(zval **)(--p);
+		zval *q = (zval *)*(--p);
 		*p = NULL;
 		zval_ptr_dtor(&q);
 	}
